### ISSUE TITLE
Lint and autofix tests

### ISSUE THREAD

**Issue by bkemper** [SparkPost/2web2ui#103]
While I am building with both `npm start` and `npm test` running, my source code gets linted and auto-fixed by Eslint after each file change and my tests rerun to confirm everything is working as expected.  As a final sanity check, I run `npm run lint` to find several linting errors.  Why?

**Comment by bkemper** [comment]
I briefly looked into this last night.  There is a reported issue, [create-react-app#1169](https://github.com/facebookincubator/create-react-app/issues/1169).  The ideal situation is linting is done with both `npm start` and `npm test`.  For now, we will have to manually run, `npm run lint -- --fix` before each commit or merge.

FWIW, I saw [this recommendation](https://github.com/facebookincubator/create-react-app/issues/1169#issuecomment-328152383) and setup [rogeliog/jest-runner-eslint](https://github.com/rogeliog/jest-runner-eslint).  It works, but I couldn't find a good way to use our Eslint configuration in package.json.

**Comment by bkemper** [comment]
I took another stab at using `jest-runner-eslint`.  After installation and setup, I continued to get a number of unexpected linting errors, primarily `no-undef` for class properties.

```
/Users/bkemper/Repos/sparkpost/2web2ui/src/pages/webhooks/DetailsPage.js
  66:3  error  'toggleDelete' is not defined  no-undef
```

Class properties require Babel support which got me thinking this isn't an eslint configuration problem rather a preprocessing problem with Babel.  I was able to eliminate Jest from the list of suspects by reproducing the issue with running eslint, `npx eslint src`.  I was able to rule out an eslint configuration problem by running `--debug` and `--print-config`.

The strange fact that eslint behaves differently when I install `jest-runner-eslint` means this has to be a packaging issue.  The following is a diff of my modules after installing `jest-runner-eslint`.

```
$ npm install --save-dev jest-runner-eslint
$ git status --short
 M package.json
?? node_modules/cosmiconfig/
?? node_modules/eslint-scope/
?? node_modules/functional-red-black-tree/
?? node_modules/jest-runner-eslint/
?? node_modules/require-from-string/
```

My money is on `eslint-scope`.  I found a related issue, [babel-eslint#487](https://github.com/babel/babel-eslint/issues/487).  I think the next step is to try upgrading eslint to v4.
